orbit/pkg/update/flag_runner.go

```diff
@@ -126,6 +126,160 @@ func (r *FlagRunner) DoFlagsUpdate() (bool, error) {
 	return true, nil
 }
 
+// ExtensionRunner is a specialized runner to periodically check and update flags from Fleet
+// It is designed with Execute and Interrupt functions to be compatible with oklog/run
+//
+// It uses an OrbitClient, along with ExtensionUpdateOptions and updateRunner to connect to Fleet
+type ExtensionRunner struct {
+	configFetcher OrbitConfigFetcher
+	opt           ExtensionUpdateOptions
+	cancel        chan struct{}
+	updateRunner  *Runner
+}
+
+// ExtensionUpdateOptions is options provided for the extensions fetch/update runner
+type ExtensionUpdateOptions struct {
+	// CheckInterval is the interval to check for updates
+	CheckInterval time.Duration
+	// RootDir is the root directory for orbit state
+	RootDir string
+}
+
+// NewExtensionConfigUpdateRunner creates a new runner with provided options
+// The runner must be started with Execute
+func NewExtensionConfigUpdateRunner(configFetcher OrbitConfigFetcher, opt ExtensionUpdateOptions, updateRunner *Runner) *ExtensionRunner {
+	return &ExtensionRunner{
+		configFetcher: configFetcher,
+		opt:           opt,
+		cancel:        make(chan struct{}),
+		updateRunner:  updateRunner,
+	}
+}
+
+// Execute starts the loop checking for updates
+func (r *ExtensionRunner) Execute() error {
+	log.Debug().Msg("starting extension runner")
+
+	ticker := time.NewTicker(r.opt.CheckInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-r.cancel:
+			return nil
+		case <-ticker.C:
+			log.Info().Msg("calling /config API to fetch/update extensions")
+			didExtensionsUpdate, err := r.DoExtensionConfigUpdate()
+			if err != nil {
+				log.Info().Err(err).Msg("ext update failed")
+			}
+			if didExtensionsUpdate {
+				log.Info().Msg("successfully updated/fetched extensions from /config API")
+				return nil
+			}
+		}
+		ticker.Reset(r.opt.CheckInterval)
```

Comment:
Why reset? Is it because the CheckInterval could change?

Reply:
We want to consistently tick at `CheckInterval` in case extension update takes longer. Not strictly needed ehre, following a similar pattern as with flags update

Reply:
To confirm I understand: The intended behavior is "each update check starts 15m (or the configured interval) after completion of the last update check." As opposed to "each update check starts 15m after the start of the prior update check".

Is that right? Either way seems fine to me, just want to make sure I understand.

Reply:
@zwass The `15m` interval (`update-interval`) is different than this `CheckInterval`.  This `CheckInterval` is configured as `60s`

This `CheckInterval` is  the interval at which orbit makes an HTTP request to `/config` endpoint to check if `extensions` have been added to `agent_options`. 

The intention here with resetting the `CheckInterval` is at we want to call `/config` API at every `CheckInterval` irrespective if the previous tick took longer for instance. A bit more discussion regarding this from a previous discussion: https://github.com/fleetdm/fleet/pull/8056#discussion_r986063310